cpp/ql/src/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -274,11 +294,32 @@ class IteratorArrayMemberOperator extends MemberFunction, TaintFunction, Iterato
   }
 }
 
+/**
+ * An `operator=` member function of an iterator class that is not a copy or move assignment
+ * operator.
+ * 
+ * The `hasTaintFlow` override provides flow through output iterators that return themselves with
+ * `operator*` and use their own `operator=` to assign to the container.
+ */
+class IteratorAssignmentMemberOperator extends MemberFunction, TaintFunction {
+  IteratorAssignmentMemberOperator() {
+    this.hasName("operator=") and
+    this.getDeclaringType() instanceof Iterator and
+    not this instanceof CopyAssignmentOperator and
+    not this instanceof MoveAssignmentOperator
+  }
+
+  override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
+    input.isParameter(0) and
```

Comment:
It's not `isParameterRef`? I'm seeing reference types in https://en.cppreference.com/w/cpp/iterator/back_insert_iterator/operator%3D